src/sgame/lua/Buildable.cpp

```diff
@@ -0,0 +1,223 @@
+/// Whether the buildable is powered.
+// @tfield boolean powered Read/Write.
+// @within Buildable
+GET_FUNC( powered, lua_pushboolean(L, bc(c->ent)->Powered() ) )
+/// The entity the buildable is targeting.
+// @tfield EntityProxy target Read only.
+// @within Buildable
+GET_FUNC( target, LuaLib<EntityProxy>::push( L, Entity::CreateProxy( c->ent->target.get(), L ) ) )
+/// Buildable health.
+// @tfield number health Read/Write.
+// @within Buildable
+GET_FUNC( health, lua_pushnumber(L, Entities::HasHealthComponent(c->ent) ? Entities::HealthOf( c->ent ) : 0 ) )
+/// Buildable team.
+// @tfield string team Read only.
+// @within Buildable
+GET_FUNC( team, lua_pushstring(L, BG_TeamName(c->ent->buildableTeam) ) )
+/// Whether this buildable is marked for deconstruction.
+// @tfield boolean marked Read only.
+// @within Buildable
+GET_FUNC( marked, lua_pushboolean(L, bc(c->ent)->MarkedForDeconstruction() ) )
+/// The level.time at which this buildable was marked.
+// @tfield integer marked_time Read only.
+// @within Buildable
+GET_FUNC( marked_time, lua_pushinteger(L, bc(c->ent)->GetMarkTime() ) )
+
+#define GETTER(name) { #name, Get##name }
+luaL_Reg BuildableGetters[] =
+{
+    GETTER( name ),
+    GETTER( powered ),
+    GETTER( target ),
+    GETTER( health ),
+    GETTER( team ),
+    GETTER( marked ),
+    GETTER( marked_time ),
+	{ nullptr, nullptr },
+};
+
+
+#define SET_FUNC( var, func) \
+static int Set##var( lua_State* L ) \
+{ \
+    Buildable* c = LuaLib<Buildable>::check(L, 1); \
+    if (!c || !c->ent || c->ent->s.eType != entityType_t::ET_BUILDABLE) \
+    { \
+        Log::Warn("trying to access stale buildable info!"); \
+        return 0; \
+    } \
+	func; \
+	return 0; \
+}
+
+SET_FUNC( powered, c->ent->powered = lua_toboolean(L, 2) )
+SET_FUNC( health, hc(c->ent)->SetHealth(luaL_checknumber(L, 2)) )
+
+static int Settarget(lua_State* L)
+{
+    Buildable* c = LuaLib<Buildable>::check(L, 1);
+    if (!c || !c->ent || c->ent->s.eType != entityType_t::ET_BUILDABLE)
+    {
+        Log::Warn("trying to access stale buildable info!");
+        return 0;
+    }
+    EntityProxy* proxy = LuaLib<EntityProxy>::check(L, 2);
+    if (!proxy || !proxy->ent)
+    {
+        c->ent->target = nullptr;
+        return 0;
+    }
+    c->ent->target = proxy->ent;
+	return 0;
+}
+
+static int Setmarked(lua_State* L)
+{
+    Buildable* c = LuaLib<Buildable>::check(L, 1);
+    if (!c || !c->ent || c->ent->s.eType != entityType_t::ET_BUILDABLE)
+    {
+        Log::Warn("trying to access stale buildable info!");
+        return 0;
+    }
+    BuildableComponent* b = bc(c->ent);
+    bool mark = lua_toboolean(L, 2);
+    if (mark != b->MarkedForDeconstruction())
+    {
+        b->ToggleDeconstructionMark();
+    }
+	return 0;
+}
+
+#define SETTER(name) { #name, Set##name }
+luaL_Reg BuildableSetters[] =
+{
+    SETTER( powered ),
+    SETTER( target ),
+    SETTER( health ),
```

Comment:
Yeah, I think that the setter for powered might not mean anything.

For target, I left it because it might be useful to encourage some functions to pick a different target (ie, trapper for example) rather than the one selected by the game.

For health, that is a possibility, but `SetHealth` is valid as well if you don't want to "damage" or "heal" as part of your action.